c7n/resources/s3.py

```diff
@@ -1052,13 +1053,49 @@ def process_resource_set(self, resource_set, tags):
                     Bucket=r['Name'], Tagging={'TagSet': tag_set})
 
 
+@actions.register('mark-for-op')
+class MarkBucketForOp(TagDelayedAction):
+
+    schema = type_schema(
+        'mark-for-op', rinherit=TagDelayedAction.schema)
+
+    def process_resource_set(self, resource_set, tags):
+        client = local_session(self.manager.session_factory).client('s3')
```

Comment:
I retested the s3 tag actions using the common tag methods. These tests resulted in errors "'NoneType' object is not callable". This is occurring because the baseline process_resource_set() for the common TagDelayedAction and RemoveTag methods are defaulted to work with ec2 resources.  in the resource/s3, I am inheriting the common TagDelayedAction & RemoveTag classes and overriding the process_resource_set() methods to work with s3. 

I could add in conditional cases to check if the aws resource is a 's3' or 'ec2', and then call their respective boto client add/remove tag functions. If we followed this approach, the common tag classes would need additional switch cases for the other aws resources, going forward.

With these considerations, which approach would you advise?